### DISCUSSION TITLE
Solving a big Delay when use a lookup for more 70,000 rows !

### DISCUSSION THREAD

**Discussion by waleed51** [serenity-is/Serenity#5413] [Help]
I have a (Visits) grid which records visits transactions to my company , 
and it have a relation with (Visitor) -Visit have one or more visitors- ,

 I use a Visitors Row as Lookup in Visit dialog, to can  select visitor when I add a new Visitor transaction, ( like in Picture).

the problem is Visitor table have more 70,000 rows,  when I use it in lookup it make a big delay for searching on it to retrieve data.

Is there is any way to not load all lookup, only load based on search in lookup, or any alternative option compatible with my business?

![Screenshot_3](https://user-images.githubusercontent.com/18492471/106300652-6e7c7f00-625f-11eb-9197-3df97d49a1e4.png)

**Comment by VictorTomaili** [comment]
Did you try the `AsyncLookupEditor`?

**Reply by waleed51** [reply]
no, i don't know it, 

plz, can you give me more explanation ??

**Reply by waleed51** [reply]
@VictorTomaili  ??

**Reply by waleed51** [reply]
I search for AsyncLookupEditor, as I understand it's not meet my business , i want all elements in lookup, i may want to search about anyone at any time,  

i think my nees is to upload lookup elements only on search in select2 => upload from database based on which one i search for

**Comment by VictorTomaili** [comment]
Duplicated #5417
